src/auto/tls_database.rs

```diff
@@ -0,0 +1,166 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ ea993ed)
+// from gir-files (https://github.com/gtk-rs/gir-files @ 469db10)
+// DO NOT EDIT
+
+use Cancellable;
+use Error;
+use SocketConnectable;
+use TlsCertificate;
+use TlsCertificateFlags;
+use TlsDatabaseLookupFlags;
+use TlsDatabaseVerifyFlags;
+use TlsInteraction;
+use ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct TlsDatabase(Object<ffi::GTlsDatabase, ffi::GTlsDatabaseClass>);
+
+    match fn {
+        get_type => || ffi::g_tls_database_get_type(),
+    }
+}
+
+pub trait TlsDatabaseExt {
+    fn create_certificate_handle(&self, certificate: &TlsCertificate) -> Option<String>;
+
+    fn lookup_certificate_for_handle<'a, 'b, P: Into<Option<&'a TlsInteraction>>, Q: Into<Option<&'b Cancellable>>>(&self, handle: &str, interaction: P, flags: TlsDatabaseLookupFlags, cancellable: Q) -> Result<Option<TlsCertificate>, Error>;
+
+    fn lookup_certificate_for_handle_async<'a, 'b, P: Into<Option<&'a TlsInteraction>>, Q: Into<Option<&'b Cancellable>>, R: FnOnce(Result<(), Error>) + Send + 'static>(&self, handle: &str, interaction: P, flags: TlsDatabaseLookupFlags, cancellable: Q, callback: R);
```

Comment:
These async functions are all wrong because they ignore the non-bool return value of the finish function (it should be put as first the component of the Result in the closure). Need to be written manually for now, see https://github.com/gtk-rs/gir/issues/535

Reply:
Congrats for sending me down a new rabbit hole :P

Reply:
You found the end of it already :)

Reply:
Yep, should be all good now :)